MetasysServices/MetasysClient.cs

```diff
@@ -1207,7 +1209,45 @@ private async Task SendCommandRequestAsync(Guid id, string command, IEnumerable<
             }
         }
 
+        #region "ad-hoc calls" // =========================================================================================================
+        // Support Ad-Hoc calls -----------------------------------------------------------------------------------------------------------
+        ///<inheritdoc/>
+        public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage requestMessage, HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead, CancellationToken cancellationToken = default)
+        {
+            var response = new HttpResponseMessage();
+            try
+            {
+                var flurlRequest = Client.Request();
+                flurlRequest.Url = GetUrlFromHttpRequest(requestMessage);
+                flurlRequest.WithHeaders(requestMessage.Headers);
+
+                response = await flurlRequest.SendAsync(requestMessage.Method, requestMessage.Content, cancellationToken, completionOption).ConfigureAwait(false);
+            }
+            catch (FlurlHttpException e)
+            {
+                ThrowHttpException(e);
+            }
+            return response;
+        }
 
+        private Url GetUrlFromHttpRequest(HttpRequestMessage requestMessage)
+        {
+            var baseUri = new Uri(Client.BaseUrl);
+            var requestUri = requestMessage.RequestUri.ToString();
+            if (Uri.IsWellFormedUriString(requestUri, UriKind.Absolute))
+            {
+                if (Uri.Compare(baseUri, requestMessage.RequestUri, UriComponents.SchemeAndServer, UriFormat.SafeUnescaped, StringComparison.OrdinalIgnoreCase) != 0)
+                {
+                    throw new MetasysHttpException("HTTP request can not be made.", "You are trying to connect to a different host.");
+                }
+                return new Url(requestUri);
+            } 
+            else
+            {
+                return new Url($"{baseUri.AbsoluteUri.Replace(baseUri.AbsolutePath, string.Empty)}/{requestUri}");

```

Comment:
I don't think combining URIs this way is a good approach. Also, how does this work? Wouldn't this expect a client to pass the `api` prefix? I think we need to be more clear in documentation of the public interface what's expected for relative paths - you are assuming they will pass a path that starts with `api/v...`. Is this what you intended? 

I think you should use classes/methods used for URI manipulation. For example `Uri.TryCreate` takes a base URI and a relative URI, and returns the result. You could also use `UriBuilder` similar to how you were before.

Reply:
> I don't think combining URIs this way is a good approach. Also, how does this work? Wouldn't this expect a client to pass the api prefix? I think we need to be more clear in documentation of the public interface what's expected for relative paths - you are assuming they will pass a path that starts with api/v.... Is this what you intended?

I thought in case of relative URI, user will provide uri starting with `api/v...`

> I think you should use classes/methods used for URI manipulation. For example Uri.TryCreate takes a base URI and a relative URI, and returns the result.

In case of relative URI the problem with `Uri.TryCreate` is it does not ensure that the path will exactly separates each segment by '/'.

I switched to flurl's `Url.Combine` because it ensures exactly one '/' separates each segment.

> You could also use UriBuilder similar to how you were before.

In case of relative URI the URI class parses the relative URI to new URI but it failed to parse Query, LocalPath & Fragment.

![image](https://user-images.githubusercontent.com/68600305/214544316-1bddf9c2-5366-4232-8e52-db50158f4011.png)